lib/Partitioner/PartitionerUtils.cpp

```diff
@@ -140,72 +140,154 @@ getOutUsersWithOnePredecessor(const std::set<Node *> &nodes) {
   return ret;
 }
 
+/// \returns the list of users who use \p node's \p idx - th output.
+std::vector<Node *> getNodeUsers(Node *node, size_t idx) {
+  std::vector<Node *> ret;
+  NodeValue nodeVal = node->getNthResult(idx);
+  for (auto &U : node->getUsers()) {
+    Node *user = U.getUser();
+    for (size_t i = 0, e = user->getNumInputs(); i < e; i++) {
+      NodeValue val = user->getNthInput(i);
+      if (val == nodeVal) {
+        ret.push_back(user);
+        break;
```

Comment:
Should this be a set instead of a vector? This could behave weird/unexpected if a node takes another node as input multiple times. For example given some node `N`, you could do `createMatMul("M", N, N)`. Then if you pass `N` into this function, it would return a vector `{N, N}`.

Reply:
The function returns the "user". In your case, the result is {Matmul_Node}. Once we detect a user, the user will be put into the result and break. So a user is only be added once.

Reply:
Whoops, sorry I had a typo. I think the result here would be `{Matmul_Node, Matmul_Node}` actually. I think `getUsers()` in general is confusing because in this case `N` would be seen as having two users instead of one, even though it's the same user.

And actually I don't think we actually need to iterate over the inputs of `user` here to find which input `node` is into `user`. We're not using that info anywhere, and it's guaranteed `user` is at least one input of `node`.